File: app/src/main/java/com/sysdevmobile/vibratedemo/MyBroadcastReceiver.java
package com.sysdevmobile.vibratedemo;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Vibrator;
import android.util.Log;

import java.util.Objects;

public class MyBroadcastReceiver extends BroadcastReceiver{

    private static final String TAG = "MyBroadcastReceiver";
    @Override
    public void onReceive(Context context, Intent intent) {
        long time;
        Log.d(TAG,""+intent.getAction());
        Vibrator v = (Vibrator) context.getSystemService(Context.VIBRATOR_SERVICE);
        if (Objects.requireNonNull(intent.getAction()).equals(context.getString(R.string.custom_intent))) {
            time = intent.getLongExtra("time", 500);
            Log.d(TAG, "onReceive: "+time);
           try {
               assert v != null;
               Objects.requireNonNull(v).vibrate(time);
           }
            catch (NullPointerException npe)
            {
                npe.printStackTrace();
            }

        }
    }
}

File: app/src/main/java/com/sysdevmobile/vibratedemo/MyBroadcastReceiver.kt
package com.sysdevmobile.vibratedemo

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.os.VibrationEffect
import android.os.Vibrator
import android.util.Log

class MyBroadcastReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val time: Long
        Log.d(TAG, "" + intent.action!!)
        if (intent.action == context.getString(R.string.custom_intent)) {
            val v = context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
            time = intent.getLongExtra("time", 500)
            Log.d(TAG, "onReceive: $time")
            try {
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                    val effect = VibrationEffect.createOneShot(time, VibrationEffect.DEFAULT_AMPLITUDE)
                    v.vibrate(effect)
                } else {
                    v.vibrate(time)
                }
            } catch (npe: NullPointerException) {
                npe.printStackTrace()
            }

        }
    }

    companion object {

        private const val TAG = "MyBroadcastReceiver"
    }
}
